### ISSUE TITLE
`linux-firmware-ath10k-tn` will not parse without `token` override

### ISSUE THREAD

**Issue by lgrosz** [TechNexion/meta-tn-wifi#1]
Hi,

## Behavior

When pulling the latest (for at least Kirkstone), the `linux-firmware-ath10k-tn` recipe does not parse without supplying the `token` override (through the `PA_TOKEN`) variable.

## Expected Behavior

The recipe should _parse_ regardless of the overrides provided. If this recipe is not needed it shouldn't stop anything else from being built.

## Workaround

Mask the problematic recipe.

**Comment by ray-chang** [comment]
@lgrosz 

Thanks for the feedback.
Because this module also contains Bluetooth, and it needs to load firmware to use BT 5.0, otherwise it will only be BT 4.2.
If you can provide a reliable link that can be downloaded to the Bluetooth firmware, we can modify it. But as far as I know, this firmware has a license, and there is no way to disclose it directly.

BR,
Ray

**Comment by lgrosz** [comment]
Would you be able to integrate something like this (untested)?

```
From 0d2121d3d1484ad363ec479c43244b44873b812a Mon Sep 17 00:00:00 2001
From: Logan Grosz <logan.grosz@gmail.com>
Date: Sun, 20 Oct 2024 22:43:46 -0600
Subject: [PATCH] linux-firmware-ath10k-tn: add to SRC_URI when token override
 present

Fixes issue where linux-firmware-ath10k-tn recipe will not parse by only
adding proprietary SRC_URI when the token override is present.
---
 recipes-kernel/linux-firmware/linux-firmware-ath10k-tn.bb | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/recipes-kernel/linux-firmware/linux-firmware-ath10k-tn.bb b/recipes-kernel/linux-firmware/linux-firmware-ath10k-tn.bb
index 6d09fbe..cc22f8a 100644
--- a/recipes-kernel/linux-firmware/linux-firmware-ath10k-tn.bb
+++ b/recipes-kernel/linux-firmware/linux-firmware-ath10k-tn.bb
@@ -10,10 +10,7 @@ LIC_FILES_CHKSUM = "file://ath10k/LICENSE.qca_firmware;md5=74852b14e2b35d8052226
 SRCREVATH10K = "ca0916244fb9ae75242585f3bb8397c5732b910c"
 SRCREVQCABT = "be98108d402e3aa76e78b978a986db052497014d"
 
-SRC_URI = "git://git.codelinaro.org/clo/ath-firmware/ath10k-firmware.git;protocol=https;branch=main;rev=${SRCREVATH10K};destsuffix=ath10k \
-           ${SRCSERVER};branch=${SRCBRANCH}${SRCOPTION};rev=${SRCREVQCABT};subpath=qca;destsuffix=qca \
-"
-
+SRC_URI = "git://git.codelinaro.org/clo/ath-firmware/ath10k-firmware.git;protocol=https;branch=main;rev=${SRCREVATH10K};destsuffix=ath10k"
 S = "${WORKDIR}"
 
 # Gitlab Personal Access Token: e.g. SbtQ_mC4fvJRA88_9jB7
@@ -22,6 +19,7 @@ TOKEN = "${@'' if (d.getVar('PA_TOKEN', True) is None or len(d.getVar('PA_TOKEN'
 SRCSERVER:token = "git://gitlab.com/technexion-imx/qca_firmware.git"
 SRCOPTION:token = ";protocol=https;user=oauth2:${TOKEN}"
 SRCBRANCH:token = "caf-wlan/CNSS.LEA.NRT_3.0"
+SRC_URI:token:append = " ${SRCSERVER};branch=${SRCBRANCH}${SRCOPTION};rev=${SRCREVQCABT};subpath=qca;destsuffix=qca"
 
 do_configure() {
         :
-- 
2.46.0
```

This should allow the recipe to parse even if the `token` override was not present, but not expose any proprietary information.

**Comment by ray-chang** [comment]
@lgrosz 

I have made the following changes according to your suggestion, because your changes are not entirely correct, so I did not follow your changes:
https://github.com/TechNexion/meta-tn-wifi/commit/b652fe3add9ffb1eb4765688ec0ed89cdddcef44

By the way, we still recommend using the firmware we provide, which we have verified. We do not guarantee that firmware from other sources will function properly.

BR,
Ray

**Comment by lgrosz** [comment]
> I have made the following changes according to your suggestion, because your changes are not entirely correct, so I did not follow your changes:

Yeah, I got the ordering of the override syntax incorrect in my patch.

> By the way, we still recommend using the firmware we provide, which we have verified. We do not guarantee that firmware from other sources will function properly.

We are just using the Wi-Fi firmware and that's even an older version that was sent to us in a tarball. We aren't supplying `PA_TOKEN` to our build at the moment. This resulted in the parsing error that caused me to bring up the issue. We are in the process of migrating our layer to a newer version and will start using `PA_TOKEN` at that time but until then it's nice the recipe will parse regardless of the environment.

Thank you! I appreciate your efforts.
